react/features/remote-video-menu/components/native/RemoteVideoMenu.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+import { connect } from 'react-redux';
+
+import {
+    BottomSheet,
+    bottomSheetItemStylesCombined,
+    hideDialog
+} from '../../../base/dialog';
+import KickButton from './KickButton';
+import MuteButton from './MuteButton';
+import styles from './styles';
+
+type Props = {
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * The participant for which this menu opened for.
+     */
+    participant: Object
+}
+
+/**
+ * The exported React {@code Component}. We need it to execute
+ * {@link hideDialog}.
+ *
+ * XXX It does not break our coding style rule to not utilize globals for state,
+ * because it is merely another name for {@code export}'s {@code default}.
+ */
+let RemoteVideoMenu_; // eslint-disable-line prefer-const
```

Comment:
You need to reference the constructor in order for hideDialog to work? Can you stack multiple dialogs so you need to target which dialog to close? Is it possible or better to reference the constructor on the instance instead (this.constructor)?

Reply:
So the reason why it's here is that the dialog is opened from another component, where it's opened using the exported reference (the result of ```connect(_mapStateToProps)(RemoteVideoMenu)``` which is not equal to ```RemoteVideoMenu```). But we try to close it here inside the component, where we only have access to ```RemoteVideoMenu``` and the ```hideDialog``` function doesn't find a match.

I reused the logic that was originally used in the OverflowMenu. However now that I rethought this approach, I found a better solution where I don't need this hack, so thanks for making me use my brain ;)